rotkehlchen/chain/evm/decoding/uniswap_v2_like/decoder.py

```diff
@@ -0,0 +1,140 @@
+            pool_address_setup: dict[ChecksumEvmAddress, str],
+    ) -> None:
+        super().__init__(evm_inquirer, base_tools, msg_aggregator)
+        self.counterparty_addresses = counterparty_addresses
+        self.pool_address_setup = pool_address_setup
+
+    def _post_decode_events(
+            self,
+            transaction: EvmTransaction,
+            decoded_events: list['EvmEvent'],
+            all_logs: list[EvmTxReceiptLog],
+    ) -> list['EvmEvent']:
+        for tx_log in all_logs:
+            self._maybe_decode_v2_swap(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                token=None,
+                action_items=[],
+                all_logs=all_logs,
+            )
+            self._maybe_decode_liquidity_addition_and_removal(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                token=None,
+                action_items=[],
+                all_logs=all_logs,
+            )
+        return decoded_events
+
+    def _maybe_decode_v2_swap(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list['EvmEvent'],
+            token: EvmToken | None,  # pylint: disable=unused-argument
+            action_items: list[ActionItem],  # pylint: disable=unused-argument
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> None:
+        """Decodes the swap if the event log matches the swap signature.
+        Does not return anything because it modifies the decoded_events list
+        """
+        if tx_log.topics[0] != SWAP_SIGNATURE:
+            return
+
+        decode_uniswap_v2_like_swap(
+            tx_log=tx_log,
+            transaction=transaction,
+            decoded_events=decoded_events,
+            counterparty=self.counterparties()[0].identifier,
+            database=self.base.database,
+            evm_inquirer=self.evm_inquirer,
+            notify_user=self.notify_user,
+            counterparty_addresses=self.counterparty_addresses,
+        )
+
+    def _maybe_decode_liquidity_addition_and_removal(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list['EvmEvent'],
+            token: EvmToken | None,  # pylint: disable=unused-argument
+            action_items: list[ActionItem],  # pylint: disable=unused-argument
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> None:
+        if tx_log.topics[0] == MINT_SIGNATURE:
+            event_action_type: Literal['addition', 'removal'] = 'addition'
+        elif tx_log.topics[0] in BURN_SIGNATURES:
+            event_action_type = 'removal'
+        else:
+            return
+
+        if self.weth_asset is None:
```

Comment:
this if does not need to exist. It's always true. If you use inheritance properly.